x-pack/plugins/enterprise_search/public/applications/app_search/components/curations/utils.ts

```diff
@@ -14,3 +14,14 @@ export const convertToDate = (serverDateString: string): Date => {
     .replace('AM', ' AM');
   return new Date(readableDateString);
 };
+
+export const addDocument = (documentArray: string[], newDocument: string) => {
+  return [...documentArray, newDocument];
+};
+
+export const removeDocument = (documentArray: string[], deletedDocument: string) => {
+  const newArray = [...documentArray];
+  const indexToDelete = newArray.indexOf(deletedDocument);
+  newArray.splice(indexToDelete, 1);
+  return newArray;
+};
```

Comment:
I dried these out because both `promotedIds `and `hiddenIds` handle their arrays in the same way. Maybe a little overengineered, but I figured why not.

Or, it might be worth checking to see if other logic files need this and pulling it out to a top-level util some day